### PR TITLE
Async request

### PR DESCRIPTION
**Pull request by artemkovalyov**
## What Has Changed?

Making sure Async requests are represented in the OData v4 docs and customers can come back to us if they need it.

## Manual Checks?

- [x] Text adheres to the style guide (`vale docs/`)
  - Every sentence is on its own line
  - Headings use [title capitalization](https://capitalizemytitle.com/style/AP/#) (applies also to the sidebar and title of a document)
  - You followed [naming center](https://www.sapbrandtools.com/naming-center/#/dashboard) guidelines when referring to SAP products (e.g. SAP S/4HANA)
- [x] You checked your spelling and grammar (consider using Grammarly, see `CONTRIBUTING.md`)
- [x] You formatted all changed files with prettier (`npm run prettier`)
- [x] You tested if the documentation still builds (`npm run build`)
- [x] You verified all new and changed links still work (changing the `id` or name of a file can break links)
~~- [] You have updated the [feature matrix](https://github.com/SAP/cloud-sdk/blob/main/docs/components/data/features.js) if you add documentation on a new feature or otherwise required.~~

### CODE REVIEW

**Review by MatKuhr — APPROVED** [review]
lgtm

consider adding a note that it is still possible to execute "normal" OData requests asynchronously == let the SDK run the synchronous request in the background

it can be achieved e.g. using `queueCallable` of our resilience API: https://sap.github.io/cloud-sdk/docs/java/features/resilience/resilience#execution-variants

**Comment by artemkovalyov** [comment]
@MatKuhr, take another look to make sure I've not confused anything. Then I'll merge.

File: docs/components/NotImplemented.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+
+const NotImplemented = () => (
+  <div className="admonition admonition-caution alert alert--warning">
```
Comment (MatKuhr) [inline docs/components/NotImplemented.js:4]: Personally I'd try not to over-use warnings, maybe a `tip` or `note` would also be sufficient here. But warning is also fine I think.

**Review by MatKuhr — APPROVED** [review]
looks great!

**Pull request merged**